operators/constellation-node-operator/controllers/nodeversion_controller.go

```diff
@@ -637,7 +646,7 @@ func (r *NodeVersionReconciler) createNewNodes(ctx context.Context, config newNo
 			requiredNodesPerScalingGroup[scalingGroupID] = outdatedNodesPerScalingGroup[scalingGroupID] - pendingJoiningNodesPerScalingGroup[scalingGroupID]
 		}
 	}
-	for scalingGroupID := range requiredNodesPerScalingGroup {
+	for _, scalingGroupID := range config.scalingGroupByID.getScalingGroupIDsSorted() {
```

Comment:
I don't quite understand why we need to sort the groups - is this an optimization? The `continue` block added below looks like it would already do the trick.

Reply:
Yeah, you're probably right. I took this from Leo's PR initially without questioning it, but thinking about it - you seem right. Will remove this.